utils/helpers/icons.js

```diff
@@ -51,7 +51,7 @@ export default (iconKey = null, data = []) => {
             }
           }
 
-          if (Object.keys(d.fields).length === counter) {
+          if (!icon && Object.keys(d.fields).length === counter) {
```

Comment:
there is no point in doing any processing if there is an icon! The whole forEach loop logic should go in if (!icon) {

}